opm/simulators/wells/WellInterfaceFluidSystem.cpp

```diff
@@ -508,19 +508,18 @@ checkMaxWaterCutLimit(const WellEconProductionLimits& econ_production_limits,
     // function to calculate water cut based on rates
     auto waterCut = [](const std::vector<double>& rates,
                        const PhaseUsage& pu) {
-
-        const double oil_rate = rates[pu.phase_pos[Oil]];
-        const double water_rate = rates[pu.phase_pos[Water]];
-
-        // both rate should be in the same direction
-        assert(oil_rate * water_rate >= 0.);
-
+        const double oil_rate = -rates[pu.phase_pos[Oil]];
+        const double water_rate = -rates[pu.phase_pos[Water]];
         const double liquid_rate = oil_rate + water_rate;
-        if (liquid_rate != 0.) {
-            return (water_rate / liquid_rate);
-        } else {
+        if (liquid_rate == 0.)
             return 0.;
-        }
+        else if (water_rate < 0)
```

Comment:
A question here, the changes here implies that we have encountered different signs for oil_rate and water_rate, is that the case?

Reply:
Yes, this assert triggers on a proprietary case. We may have situations where producers temporary injects epsilon during the iterations process.